src/cdk/dialog/dialog.spec.ts

```diff
@@ -125,6 +126,21 @@ describe('Dialog', () => {
     );
   });
 
+  it('should use custom test injector', () => {
```

Comment:
The test can be simplified to this which would allow for the two helper classes to be removed:

```
it('should use custom injector', () => {
  const token = new InjectionToken<string>('message');
  const injector = Injector.create({providers: [{provide: token, useValue: 'hello'}]});
  const dialogRef = dialog.open(PizzaSnack, {injector});
  viewContainerFixture.detectChanges();

  expect(dialogRef.componentInstance?.injector.get(token)).toBe('hello');
});
```